### PR TITLE
Re-enable vibrancy and add support for Linux

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Fixes support for vibrancy
  - Disable background when vibrancy is enabled
    - Minor refactors to pass the relevant config option to dashboard, so that the dashboard knows whether vibrancy is enabled
  - Enable vibrancy on Linux if a special environment variable is set

### Important Notes
- MUST be tested on macOS
- Windows *does not support resizing* when vibrancy is enabled. This is a limitation of Windows - see [this issue](https://github.com/electron/electron/issues/6107).

### Screenshots
:information_source: Because the background is transparent, `backdrop-blur` does not fully obscure the text behind semi-transparent elements.

- Linux, no DE, blur enabled in picom
  - This is my normal setup, transparency is permanently enabled for unfocused windows (see devtools), but Enso is focused on this screenshot so picom's transparency is disabled.
image

- Windows, window focused:
  - Notice that the maximize button is absent. The window resize handles are also not available.
image

- Windows, window unfocused:
  - Vibrancy *does not work* on windows if the window is not focused.
image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

**Comment by vitvakatu** [comment]
Do we have a task for this? I don’t understand what the `vibrancy` effect means, and searching online doesn’t help either. Please attach the screenshots.

File: app/ide-desktop/lib/client/src/index.ts [hunk 1]
```diff
@@ -279,7 +279,8 @@ class App {
                 const useFrame = this.args.groups.window.options.frame.value
                 const macOS = process.platform === 'darwin'
                 const useHiddenInsetTitleBar = !useFrame && macOS
-                const useVibrancy = this.args.groups.window.options.vibrancy.value
+                const useVibrancy = (this.args.groups.window.options.vibrancy.value &&=
```
Comment (Frizi) [inline app/ide-desktop/lib/client/src/index.ts]: Is the assignment here intentional? If so, I’d split that into separate lines.

**Review by Frizi — APPROVED** [review]

File: app/ide-desktop/lib/client/src/index.ts (hunk 1 above)
Comment (somebody1234) [reply app/ide-desktop/lib/client/src/index.ts]: thanks, fixed

**Comment by somebody1234** [comment]
@vitvakatu We don't have a task for this, but the option has existed for quite a long time, and the background image that was added last year effectively disabled it, so I figured it makes sense to bring it back.

Vibrancy is indeed a very niche term, [used mostly by macOS](https://developer.apple.com/design/human-interface-guidelines/materials).
Windows refers to this effect as the [Acrylic](https://learn.microsoft.com/en-us/windows/apps/design/style/acrylic) material, part of the Fluent design system.
I believe Linux doesn't have its own term for it, but there is a [Fluent](https://store.kde.org/p/1280231) theme.
In general though it seems to be more reasonably named as [blur](https://store.kde.org/find?search=blur&f=tags)...

**Comment by vitvakatu** [comment]
I don’t see any particular effect on MacOS, do I need to enable it somewhere?

**Comment by somebody1234** [comment]
@vitvakatu ah, my bad. yeah it has been turned off by default for a long while now. it needs to be explicitly enabled by passing `-window.vibrancy` as a CLI parameter

**Comment by vitvakatu** [comment]
Ok, here are screenshots from Mac:

Screenshot 2024-02-09 at 4 04 24 PM
Screenshot 2024-02-09 at 4 04 44 PM

Also `-window.vibrancy` flag is not visible in `--help` output.

**Comment by somebody1234** [comment]
hmm, is the titlebar missing? that might be an issue on mac, right?

as for it not being visible in `--help` output... i'm assuming it's intentional for now (?) - afaict (kinda like dark theme) it's not something we want to *officially* support currently? 

(that said, i'm not quite sure whether the theme option still exists)

**Comment by vitvakatu** [comment]
> hmm, is the titlebar missing? that might be an issue on mac, right?

Oh, it actually isn’t, I just cropped screenshots a bit. It also works for currently focused window only.

Screenshot 2024-02-09 at 4 42 27 PM

**Review by indiv0 — APPROVED** [review]
LGTM, QA passed ✅

**Pull request merged**